core/drivers/atmel_shdwc_a32.S

```diff
@@ -0,0 +1,59 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2015 Atmel Corporation,
+ *                    Nicolas Ferre <nicolas.ferre@atmel.com>
+ * Copyright (c) 2021, Microchip
+ */
+
+
+#include <arm.h>
+#include <arm32_macros.S>
+#include <asm.S>
+#include <at91_pmc.h>
+#include <drivers/atmel_shdwc.h>
+#include <drivers/sam/at91_ddr.h>
+
+/**
+ * Shutdown the CPU
+ *
+ * This function is in assembly to be sure the code fits in a single cache line.
+ * We are going to power down the SDRAM and thus we can't fetch code from it
+ * once powered down.
+ *
+ * r0 = mpddrc_base
+ * r1 = shdwc_base
+ * r2 = pmc_base
+ */
+FUNC atmel_shdwc_shutdown_asm , :
+
+	ldr r3, =AT91_DDRSDRC_LPDDR2_PWOFF
+	ldr r4, =(AT91_SHDW_KEY | AT91_SHDW_SHDW)
+
+/* Align to cache line */
+.balign 32
+	/* Ensure AT91_SHDW_CR is in the TLB by reading it */
+	ldr	r6, [r1, #AT91_SHDW_CR]
+
+	/* Power down SDRAM0 */
```

Comment:
Well, maybe power down...

Reply:
`/* Power down SDRAM0 if mpddrc_base is set */`